XamarinCommunityToolkit/Core/MediaSource.shared.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using Xamarin.CommunityToolkit.Helpers;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Core
+{
+	public abstract class MediaSource : Element
+	{
+		readonly WeakEventManager weakEventManager = new WeakEventManager();
+
+		public static MediaSource FromFile(string file) =>
+			new FileMediaSource { File = file };
+
+		public static MediaSource FromUri(Uri uri)
+		{
+			if (!uri.IsAbsoluteUri)
+				throw new ArgumentException("uri is relative");
+
+			return new UriMediaSource { Uri = uri };
+		}
+
+		public static implicit operator MediaSource(string source) =>
+			Uri.TryCreate(source, UriKind.Absolute, out var uri) && uri.Scheme != "file" ? FromUri(uri) : FromFile(source);
+
+		public static implicit operator MediaSource(Uri uri)
+		{
+			if (uri == null)
+				return null;
+			if (!uri.IsAbsoluteUri)
+				throw new ArgumentException("uri is relative");
+
```

Comment:
Do we need to perform these checks here, since ```!uri.IsAbsoluteUri``` is performed inside ```FromUri``` (btw, in ```FromUri`` we don't check Uri for null)

Reply:
removed this check